Decred Wallet/Features/Receive/ReceiveViewController.swift

```diff
@@ -0,0 +1,179 @@
+        DispatchQueue.main.async {
+            self.walletAddressLabel.text = receiveAddress
+            self.displayQRCodeImage(for: receiveAddress)
+        }
+    }
+
+    private func generateNewAddress() {
+        self.oldAddress = self.walletAddressLabel.text!
+        self.getNextAddress()
+    }
+
+    @objc private func getNextAddress() {
+        if let wallet = self.selectedWallet,
+            let account = self.selectedAccount {
+                let receiveAddress = wallet.nextAddress(account.number, error: nil)
+                if self.oldAddress != receiveAddress {
+                    self.displayAddressAndQRCode(receiveAddress: receiveAddress)
+                } else if receiveAddress != "" {
+                    self.getNextAddress()
+                }
+        }
+    }
+
+    private func displayQRCodeImage(for address: String) {
+        guard let colorFilter = CIFilter(name: "CIFalseColor") else {
+            self.addressQRCodeImageView.image = nil
+            return
+        }
+
+        let filter = CIFilter(name: "CIQRCodeGenerator")
+        filter?.setValue(address.utf8Bits, forKey: "inputMessage")
+
+        colorFilter.setDefaults()
+        colorFilter.setValue(filter!.outputImage, forKey: "inputImage")
+        colorFilter.setValue(CIColor.black, forKey: "inputColor0")
+        colorFilter.setValue(CIColor.clear, forKey: "inputColor1")
+
+        if let qrImage = colorFilter.outputImage {
+            let frame = self.addressQRCodeImageView.frame
+            let smallerSide = frame.size.width < frame.size.height ? frame.size.width : frame.size.height
+            let scale = smallerSide/qrImage.extent.size.width
+            let transformedImage = qrImage.transformed(by: CGAffineTransform(scaleX: scale, y: scale))
+            self.addressQRCodeImageView.image =  UIImage(ciImage: transformedImage)
+        } else {
+            self.addressQRCodeImageView.image = nil
+        }
+    }
+
+    @IBAction func onClose(_ sender: Any) {
+        self.dismissView()
+    }
+
+    @IBAction func infoMenuButtonTapped(_ sender: Any) {
+        DispatchQueue.main.async {
+            let alertController = UIAlertController(title: LocalizedStrings.receiveDCR,
+                                                    message: LocalizedStrings.receiveInfo,
+                                                    preferredStyle: UIAlertController.Style.alert)
+            alertController.addAction(UIAlertAction(title: LocalizedStrings.gotIt,
+                                                    style: UIAlertAction.Style.default,
+                                                    handler: nil))
+            self.present(alertController, animated: true, completion: nil)
+        }
+    }
+
+    func updateSelectedAccount(_ selectedWalletId: Int, _ selectedAccount: DcrlibwalletAccount) {
+        guard let wallet = WalletLoader.shared.multiWallet.wallet(withID: selectedWalletId), wallet.isSynced()
```

Comment:
Should be `!wallet.isRestored || wallet.isSynced()`, we should allow unsynced wallets if they are not restored wallets.